src/transformers/pipelines/fill_mask.py

```diff
@@ -115,25 +115,53 @@ def __call__(self, *args, targets=None, top_k: Optional[int] = None, **kwargs):
         inputs = self._parse_and_tokenize(*args, **kwargs)
         outputs = self._forward(inputs, return_tensors=True)
 
+        # top_k must be defined
+        if top_k is None:
+            top_k = self.top_k
+
         results = []
         batch_size = outputs.shape[0] if self.framework == "tf" else outputs.size(0)
 
         if targets is not None:
-            if len(targets) == 0 or len(targets[0]) == 0:
-                raise ValueError("At least one target must be provided when passed.")
             if isinstance(targets, str):
                 targets = [targets]
 
-            targets_proc = []
+            vocab = self.tokenizer.vocab
```

Comment:
Not all tokenizers have the `vocab` property, for example the slow `RobertaTokenizer` doesn't have it

Reply:
Is there any way to get reliably a vocab (or reverse vocab) ? That undermines the purpose of this PR which is to add fast lookup. `get_vocab()` exists and is mandatory for all fast tokenizers for instance.

In this case we don't really need to `tokenize` but really just a way to get corresponding ids to reduce the potential targets.
The rationale here is:
- Fast lookup
- If lookup fail, slow tokenization and take first token. It will enable more "natural" usage but still most likely slightly wrong.

If "something" does not correspond to a single token ("some", + "thing") but is in the target list, then we will filter on "some" and it will be a possible output of the fill-mask which is .. subpar but best effort.

Reply:
@LysandreJik just checked, `get_vocab` seems like supported everywhere, isn't it ? (It's at least the case for Roberta)

I can still try and fail to err on the side of caution.

Reply:
Yes, indeed, the `get_vocab` is tested for all tokenizers! Let's go with that one.